src/Unirest/Request.php

```diff
@@ -471,9 +569,28 @@ public static function send($method, $url, $body = null, $headers = array(), $us
             ));
         }
 
-        $response   = curl_exec(self::$handle);
-        $error      = curl_error(self::$handle);
-        $info       = self::getInfo();
+        $retryCount        = 0;                           // current retry count
+        $waitTime          = 0.0;                         // wait time in secs before current api call
+        $allowed_wait_time = self::$maximumRetryWaitTime; // remaining allowed wait time in seconds
+        do {
+            // If Retrying i.e. retryCount >= 1
+            if ($retryCount > 0) {
+                usleep($waitTime * 1000000);
```

Comment:
Note: Values larger than 1000000 (i.e. sleeping for more than a second) may not be supported by the operating system. Conditionally use `sleep()` instead when the value is larger than 1000000.

Source: https://www.php.net/manual/en/function.usleep.php

Reply:
Fixed by using sleep with the combination of usleep